vendor/k8s.io/kubernetes/pkg/volume/glusterfs/glusterfs.go

```diff
@@ -676,7 +696,10 @@ func (p *glusterfsVolumeProvisioner) CreateVolume(gid int) (r *api.GlusterfsVolu
 		glog.Errorf("glusterfs: failed to create glusterfs rest client")
 		return nil, 0, fmt.Errorf("failed to create glusterfs REST client, REST server authentication failed")
 	}
-	clusterIds := dstrings.Split(p.clusterId, ",")
+	if p.provisioningConfig.clusterId != "" {
+		clusterIds = dstrings.Split(p.clusterId, ",")
+		glog.V(4).Infof("glusterfs: provided clusterids: %v", clusterIds)
+	}
 	gid64 := int64(gid)
 	volumeReq := &gapi.VolumeCreateRequest{Size: sz, Clusters: clusterIds, Gid: gid64, Durability: gapi.VolumeDurabilityInfo{Type: durabilityType, Replicate: gapi.ReplicaDurability{Replica: replicaCount}}}
```

Comment:
@screeley44 no. I am still not convinced on that fix. We will have a discussion on this and will act accordingly.